charts/shoot-core/components/charts/coredns/templates/coredns-deployment.yaml

```diff
@@ -13,6 +13,7 @@ spec:
     type: RollingUpdate
     rollingUpdate:
       maxUnavailable: 0
+      maxSurge: 1
```

Comment:
Why did you limit the `maxSurge`? As it will take longer with the extended initial readiness delay to get a core dns pod ready, this change additionally prolongs the roll-out of a change. Is that intentional? If yes, why?

Reply:
Yes, it is intentional. However, there is no actual difference as the default is `25%` while the HPA is configured to have maximum replicas `5`. The reason is that there might be applications deployed in the clusters that are sensitive to coredns unavailability (even less several seconds) and in this case it is better to slowly rollout the deployment and give the new replicas enough time to initialize their caches.

Reply:
Thanks for the explanation.